core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java

```diff
@@ -280,9 +280,9 @@ protected void startServingRPCServer() {
       GrpcServerBuilder serverBuilder = GrpcServerBuilder.forAddress(mRpcBindAddress,
           ServerConfiguration.global());
 
-      mRPCExecutor =
-          new ForkJoinPool(ServerConfiguration.getInt(
-              PropertyKey.MASTER_RPC_FORKJOIN_POOL_PARALLELISM));
+      mRPCExecutor = Executors.newFixedThreadPool(ServerConfiguration.getInt(
```

Comment:
You can use mMinWorkerThreads/mMaxWorkerThreads member variables of the base MasterProcess. 

Default supplier for MASTER_WORKER_THREADS_MAX key will need to be modified though, it currently returns around 30K in linux. We don't want a high upper bound with gRPC.

Reply:
Looks like the default supplier for `MASTER_WORKER_THREADS_MAX` returns 2048 or max file descriptor/3, should we just set it to a constant like 500?

Reply:
For now it's fine to set it to a constant. We would eventually need to come up with a value that is calculated with core-count/memory into account. That'll need further profiling with various machine configurations.

Reply:
The reason for this limitation being; 
(max file descriptor/3) returns a very big number(~30K) and executors don't cool off as they used to with thrift, so thread count ends up rapidly climbing up to a high bound.